haystack_experimental/utils/jinja_chat_extension.py

```diff
@@ -0,0 +1,203 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import Any, Callable, List, Optional, Union, get_args
+
+from haystack import logging
+from jinja2 import TemplateSyntaxError, nodes
+from jinja2.ext import Extension
+
+from haystack_experimental.dataclasses.chat_message import (
+    ChatMessage,
+    ChatMessageContentT,
+    ChatRole,
+    TextContent,
+    _deserialize_content_part,
+    _serialize_content_part,
+)
+
+logger = logging.getLogger(__name__)
+
+START_TAG = "<haystack_content_part>"
+END_TAG = "</haystack_content_part>"
+
+class ChatMessageExtension(Extension):
+    """
+    A Jinja2 extension for creating structured chat messages with mixed content types.
+
+    This extension provides a custom {% message %} tag that allows creating chat messages
+    with different attributes (role, name, meta) and mixed content types (text, images, etc.).
+
+    Inspired by [Banks](https://github.com/masci/banks).
+
+    Example:
+    {% message role="system" %}
+    You are a helpful assistant. You like to talk with {{user_name}}.
+    {% endmessage %}
+
+    {% message role="user" %}
+    Hello! I am {{user_name}}. Please describe the images.
+    {% for image in images %}
+    {{ image | for_template }}
+    {% endfor %}
+    {% endmessage %}
+
+    ### How it works
+    1. The {% message %} tag is used to define a chat message.
+    2. The message can contain text and other structured content parts.
+    3. To include a structured content part in the message, the `| for_template` filter is used.
```

Comment:
Also more of just a question. Why the name `for_template`? I'm not yet quite understanding what it does based on just the name.

Reply:
Good point. I would like a short but descriptive name.
Other ideas: `content_part`/`prepare_content_part`/`mark_content_part`/`format_content_part`; something related to "including".

Do you have any other ideas?
(We can also involve some other team members if necessary)

Reply:
Yeah I'm not entirely sure either. I'd be happy to have other team members give their input